docs/src/main/asciidoc/security-openid-connect-providers.adoc

```diff
@@ -406,6 +406,42 @@ quarkus.oidc.token.customizer-name=azure-access-token-customizer
 
 ====
 
+[[slack]]
+=== Slack
+
+Create a https://api.slack.com/authentication/sign-in-with-slack#setup[Slack application]:
+
+image::oidc-slack-1.png[role="thumb"]
+
+Select application name, workspace, and remember it, you will need it later:
+
+image::oidc-slack-2.png[role="thumb"]
+
+Please save client id and secret displayed on the next page, you will need them later:
+
+image::oidc-slack-3.png[role="thumb"]
+
+Configure redirect URLs.
+Slack provider requires HTTPS protocol, for development purposes, you can use ngrok:
+
+image::oidc-slack-4.png[role="thumb"]
+
+You can now configure your `application.properties`:
+
+[source,properties]
+----
+quarkus.oidc.provider=slack
+quarkus.oidc.client-id=<Client ID>
+quarkus.oidc.credentials.secret=<Client Secret>
+quarkus.oidc.authentication.extra-params.team=quarkus-slack <1>
+----
+<1> Replace team name `quarkus-slack` with a name you specified earlier.
```

Comment:
```suggestion
<1> Use the `team` parameter to refer to the workspace you chose during the Slack OIDC application registration. 
```

What I'm not sure about, what exactly does it really mean to request an access to the workspace like `quarkus-slack`, what does it give us ? I.e, is `team` optional or required parameter ? I thought I could use it for example, if I want to check message from my company's Slack organization for example, but if someone just wants to authenticate the user with Slack, then it is probably not required...

Reply:
That's terminology used by Slack, I am not 100 % sure either, but yeah, I think it is required. For example, you need to create Slack application under some workspace (even though you can publish it later). I tried it with internal RH and I didn't had perms.

Here it is documented https://api.slack.com/authentication/sign-in-with-slack#setup and it says you can distribute the app to other workspaces, but in my understanding, it still exist under some workspace. So you cannot have app without workspace and if you check `docs/src/main/asciidoc/images/oidc-slack-5.png` it also says workspace.

I don't want to mislead you, but that is how I understand it.

> I thought I could use it for example, if I want to check message from my company's Slack organization, but if someone just wants to authenticate the user with Slack, then it is probably not required...

I  believe that is determined by scopes, I tried it without any user scopes and you were completely right, you don't need additional. Sorry I misread docs before and thanks for correcting me. Anyway, in order to sign in, you need this application and that application exist in some workspaces. Please check `docs/src/main/asciidoc/images/oidc-slack-4.png`.

Reply:
I applied your suggestion.

Reply:
Thanks @michalvavrik, can you give me a favor and name the workspace as `Quarkus`, when you get a few mins ? That should read better, for ex, `quarkus-slack is requesting a permission to access Quarkus Slack workspace`, as having `quarkus-slack` listed twice might be a bit confusing...

Reply:
np, updated, but moment, I forgot the red lines, I'll fix that

Reply:
should be done